codegen-sbt/src/main/scala/caliban/codegen/CompileTimeCalibanPlugin.scala

```diff
@@ -0,0 +1,450 @@
+                  sbt.IO.writeLines(
+                    file = metadataFile,
+                    lines = generated.map { case (f, (a, b)) => s"${f.getAbsolutePath}#$a#$b" },
+                    charset = defaultCharset,
+                    append = false
+                  )
+                  fsync(metadataFile.toPath)
+
+                  // Mainly useful for tests
+                  val touchFile: File = file(s"$metadataDir/touch")
+                  sbt.IO.touch(touchFile, setModified = true)
+                  fsync(touchFile.toPath)
+
+                  generated.map(_._1)
+                }
+
+              /**
+               * These settings are used to track the need to re-generate the code.
+               *
+               * When one of the value of these settings changes, then this plugin knows that it has to re-generate the code.
+               */
+              val trackedSettings: Def.Initialize[Task[TrackedSettings]] =
+                Def.taskDyn {
+                  Def.task {
+                    TrackedSettings(
+                      List(
+                        caliban.codegen.BuildInfo.version,
+                        zio.BuildInfo.version,
+                        pluginSettings.mkString
+                      )
+                    )
+                  }
+                }
+
+              cached("ctCalibanServer", trackedSettings)(generateSources)
+            }
+          }.value
+      )
+    )
+
+  override lazy val projectSettings: Seq[Def.Setting[_]] =
+    Seq(
+      libraryDependencies += "com.github.ghostdogpr" %% "caliban-tools" % BuildInfo.version % Compile,
+      (Compile / sourceGenerators) += Compile / ctCalibanServer / ctCalibanServerGenerate,
+      (Test / sourceGenerators) += Test / ctCalibanServer / ctCalibanServerGenerate
+    ) ++ inConfig(Compile)(pluginSettings) ++ inConfig(Test)(pluginSettings)
+}
+
+/**
+ * Sbt plugin authors documentation:
+ * ---------------------------------
+ *
+ * Interesting documentations about how to write sbt plugins:
+ *  - https://www.scala-sbt.org/1.x/docs/Plugins.html
+ *  - https://www.scala-sbt.org/1.x/docs/Plugins-Best-Practices.html
+ */
+object CompileTimeCalibanClientPlugin extends AutoPlugin {
+  override def requires = plugins.JvmPlugin
+  override def trigger  = noTrigger
+
+  private def helpMsg: String =
+    """
+      |Missing configuration for the "CompileTimeCalibanClientPlugin" plugin.
+      |
+      |You need to configure the `Compile / ctCalibanClient / ctCalibanClientsSettings` setting.
+      |
+      |Here's an example of configuration:
+      |```
+      |lazy val server =
+      |  project
+      |    .in(file("modules/server"))
+      |    .enablePlugins(CompileTimeCalibanServerPlugin)
+      |    .settings(
+      |      Compile / ctCalibanServer / ctCalibanServerSettings ++=
+      |        Seq(
+      |          "io.example.reference.GraphQLApi.api" -> ClientGenerationSettings.default,
```

Comment:
Same here

Reply:
Same answer as previously.

For example, when writing my tests, I discovered a deadlock in my code thanks to these configurations.

Reply:
Done